File: TESTING/schema/collection.schema.ts
import mongoose, { Schema } from "mongoose";

export interface ICollection {
  name: string;
  _id: string;
  description: string;

  thumnail?: string;

  icon?: string;

  isPublic: boolean;

  language: string;

  cards?: string[];

  owner: string | Schema.Types.ObjectId; // Reference to the User schema
}

export const collectionSchema = new Schema<ICollection>(
  {
    name: { type: String, unique: true },

    description: String,
    thumnail: String,
    icon: String,
    isPublic: { type: Boolean, default: true },
    language: { type: String, default: "en" },
    cards: [{ type: Schema.Types.ObjectId, ref: "Flashcard" }],
    owner: { type: Schema.Types.ObjectId, ref: "User" }, // Reference to the User schema
  },
  {
    timestamps: true,
  }
);

export const Collection = mongoose.model("collections", collectionSchema);

File: test/test-database-go/pkg/mongodb/schema/Collection.schema.go
package schema

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

// Collection represents the MongoDB collection document
type Collection struct {
    ID          primitive.ObjectID   `bson:"_id,omitempty"`        // MongoDB ObjectID
    Name        string               `bson:"name"`                 // Required and unique field
    Description string               `bson:"description,omitempty"`// Optional field
    Thumbnail   string               `bson:"thumbnail,omitempty"`  // Optional field (note the typo fix from "thumnail")
    Icon        string               `bson:"icon,omitempty"`       // Default value "Icon" (but handled in the code, not struct)
    IsPublic    bool                 `bson:"isPublic"`             // Default value true
    Language    string               `bson:"language"`             // Default value "en"
    Cards       []primitive.ObjectID `bson:"cards"`                // Array of references to FlashCards (ObjectID references)
    Owner       primitive.ObjectID   `bson:"owner"`                // Reference to the User schema (ObjectID)
}

